### PR TITLE
feat: homepage, authentication functionality

### PR DESCRIPTION
**Pull request by mariyasada**
-homepage enhancement
- an added login page, signup page, and its functionality.
- when the user successfully logged in its redirects to the homepage.
login
homepage
signup

### CODE REVIEW

File: src/contexts/auth-context.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+const initialAuthState={
+    isLoggedIn:token?true:false,
+    authenticationToken:token,
+}
+const AuthProvider=({children})=>{
+    const [user,setUser]=useState(initialAuthState);
+    const navigate=useNavigate();
+
+    const logInHandler= async(logInData)=>{
+        console.log(logInData);
```
Comment (Bhakti199) [inline src/contexts/auth-context.js]: remove console.log

File: src/constants/auth-Constants.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+export const initialSignUpData = {
+    firstName: "",
+    lastName: "",
+    username: "",
+    password: ""
+}
+export const guestData = {
+    username: "adarshbalika",
+    password: "adarshBalika123",
+  };
```
Comment (Bhakti199) [inline src/constants/auth-Constants.js:14]: liked this.

**Review by Bhakti199 — APPROVED** [review]
- This code looks fine to me.
- liked the choices of images.
- remove unwanted console.log
- overall code looks good to me.
- This pr is good to merge.

File: src/Services/Authservices/loginservices.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import axios from "axios";
+
+export const logInService=async(logInData)=>{    
+    try{
+      const {data,status}=await axios.post("/api/auth/login",logInData)
+            
+            console.log(data,status) ;
```
Comment (t007rushi) [inline src/Services/Authservices/loginservices.js]: console to remove

File: src/index.js [hunk 4]
```diff
@@ -3,13 +3,19 @@ import ReactDOM from "react-dom";
...
+import { CombineProvider } from "./contexts/combineProvider";
 
 // Call make Server
 makeServer();
 
 ReactDOM.render(
   <React.StrictMode>
-    <App />
+    <Router>
+      <CombineProvider>
```
Comment (t007rushi) [inline src/index.js:15]: Nice that you made Combined Provider

File: src/pages/Home/Home.css [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  border: none;
+}
+@media screen and (max-width: 900px) {
+  .home-page {
+    flex-direction: column;
+    justify-content: center;
+    gap: 0;
+  }
+  .image-container img {
+    /* height: 79vh; */
```
Comment (t007rushi) [inline src/pages/Home/Home.css]: comments left

**Review by t007rushi — APPROVED** [review]
Good work keep it Up

**Pull request merged**